Flickering screen when moving the location of a repo

My screen flickers when moving the location of a repo locally. This was previously reported https://github.com/desktop/desktop/issues/6670. 

Reproduce:
- select a repo
- open in finder
- drag it to your local desktop

Logs:
```
install.ts:27 Error refreshing pull requests for '1121212'
Error: Unable to find path to repository on disk.
    at /webpack:/app/node_modules/dugite/build/lib/git-process.js:87:1
    at exithandler (child_process.js:294:5)
    at ChildProcess.errorhandler (child_process.js:306:5)
    at emitOne (events.js:116:13)
    at ChildProcess.emit (events.js:211:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:196:12)
    at onErrorNT (internal/child_process.js:372:16)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
```

Note I am on 1.6.2-beta2, mac and this is not reproducible on the latest prod. Also, once the repo has manually placed back it was not "activated" again automatically as expected. I had to click the Locate button and re-add it.

--- a/app/src/lib/stores/app-store.ts
+++ b/app/src/lib/stores/app-store.ts
@@ -1943,13 +1943,14 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return repository
     }
 
-    let foundRepository = await pathExists(repository.path)
+    const foundRepository =
+      (await pathExists(repository.path)) &&
+      (await isGitRepository(repository.path))
 
     if (foundRepository) {
-      foundRepository = await isGitRepository(repository.path)
+      this._updateRepositoryMissing(repository, false)
     }
-
-    return this._updateRepositoryMissing(repository, foundRepository)
+    return repository
   }
 
   /** This shouldn't be called directly. See `Dispatcher`. */

